File: src/main/java/util/Platform.java
package util;

import lombok.Getter;
import org.apache.commons.lang3.SystemUtils;

@Getter
public enum Platform {
  WINDOWS("Windows"),
  LINUX("Linux"),
  MAC("Mac"),
  UNKNOWN("Unknown");

  final String fullName;
  final String displayName;

  Platform(String displayName) {
    this.displayName = displayName;
    fullName = System.getProperty("os.name");
  }

  public static Platform detectPlatform() {
    if (SystemUtils.IS_OS_WINDOWS) {
      return WINDOWS;
    } else if (SystemUtils.IS_OS_LINUX) {
      return LINUX;
    } else if (SystemUtils.IS_OS_MAC) {
      return MAC;
    }
    return UNKNOWN;
  }
}

File: src/main/java/util/Platform.kt
package util

import lombok.Getter
import org.apache.commons.lang3.SystemUtils

/**
 * Enum representing different platforms on which the application can run.
 * Provides utility methods to detect the current operating system.
 */
@Getter
enum class Platform(val displayName: String) {
    WINDOWS("Windows"), // Represents the Windows operating system
    LINUX("Linux"), // Represents the Linux operating system
    MAC("Mac"), // Represents the macOS operating system
    UNKNOWN("Unknown"); // Represents an unknown operating system

    val fullName: String = System.getProperty("os.name") // The full name of the operating system

    companion object {
        /**
         * Detects the current platform based on system properties.
         *
         * @return The detected platform as a [Platform] enum.
         */
        fun detectPlatform(): Platform {
            return when {
                SystemUtils.IS_OS_WINDOWS -> WINDOWS
                SystemUtils.IS_OS_LINUX -> LINUX
                SystemUtils.IS_OS_MAC -> MAC
                else -> UNKNOWN
            }
        }
    }
}